core/lib/spree/core/engine.rb

```diff
@@ -44,14 +44,29 @@ class Engine < ::Rails::Engine
         Migrations.new(config, engine_name).check
       end
 
-      # Setup Event Subscribers
-      initializer 'spree.core.initialize_subscribers' do |app|
-        app.reloader.to_prepare do
-          Spree::Event.activate_autoloadable_subscribers
-        end
+      # Setup pub/sub
+      initializer 'spree.core.pub_sub' do |app|
+        if Spree::Config.use_legacy_events
+          app.reloader.to_prepare do
+            Spree::Event.activate_autoloadable_subscribers
+          end
+
+          app.reloader.before_class_unload do
+            Spree::Event.deactivate_all_subscribers
+          end
+        else
+          app.reloader.to_prepare do
+            Spree::Bus.clear
```

Comment:
❓ Does this change anything with regards to where applications or Solidus extensions should be registering any custom events they want to introduce that use the `Spree::Bus` 🤔

Reply:
That's a great question. More than a change, it's a fix. To survive code reloading, events need to be registered, and subscriptions must be added within a `#to_prepare` block (which runs one on production but at every reload on development). Solidus owns `Spree::Bus`, so it clears it at every reload. Then, solidus itself, extensions, and applications can just call `#register` & `#subscribe_to` within their own `#to_prepare` blocks, which will run later in time.